go/arrow/internal/arrdata/ioutil.go

```diff
@@ -172,6 +173,47 @@ func WriteFile(t *testing.T, f *os.File, mem memory.Allocator, schema *arrow.Sch
 	}
 }
 
+// WriteFile writes a list of records to the given file descriptor, as an ARROW file.
+func WriteFileCompressed(t *testing.T, f *os.File, mem memory.Allocator, schema *arrow.Schema, recs []array.Record, codec flatbuf.CompressionType) {
+	t.Helper()
+
+	opts := []ipc.Option{ipc.WithSchema(schema), ipc.WithAllocator(mem)}
+	switch codec {
+	case flatbuf.CompressionTypeLZ4_FRAME:
+		opts = append(opts, ipc.WithLZ4())
+	case flatbuf.CompressionTypeZSTD:
+		opts = append(opts, ipc.WithZstd())
+	}
```

Comment:
fail or at least warn on behavior for unrecognized type?

Reply:
sure i'll add a failure here for that.

Reply:
Seems like errors in this method are returned via `t`, Should the same be done there?

Reply:
ah fair point, this is a testing function I should just use the `t`. I'll fix that

Reply:
fixed